core/capabilities/webapi/round_robin_selector.go

```diff
@@ -14,11 +15,27 @@ type RoundRobinSelector struct {
 	mu    sync.Mutex
 }
 
-func NewRoundRobinSelector(items []string) *RoundRobinSelector {
-	return &RoundRobinSelector{
+func WithFixedStart() func(*RoundRobinSelector) {
+	return func(rrs *RoundRobinSelector) {
+		rrs.index = 0
+	}
+}
+
```

Comment:
It would be nice to have some inline documentation on how this works, e.g.
```suggestion
// NewRoundRobinSelector creates a selector that will select an item from a list using the round robin strategy.
// By default the index starts on a random start point in the list.
// Use the WithFixedStart modifier to change the start index.
```